libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4024,18 +4041,43 @@ void TypeChecker::endVisit(Literal const& _literal)
 					// This was already validated in visit(Literal) but the error is not fatal.
 					if (!mantissa || !exponent)
 						solAssert(!m_errorReporter.errors().empty());
-					else if (
-						!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)) ||
-						!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1))
-					)
-						// TODO: Is this triggered when the argument is out of range? Test.
-						parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+					// TODO: Is this triggered when the argument is out of range? Test.
+					else if (!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
+						parameterTypeMessage = fmt::format(
+							"Type {} of the mantissa cannot be converted to type {} of the suffix function parameter.",
+							mantissa->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(0)->toString(true)
+						);
+					else if (!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1)))
+					{
+						if (!parameterTypeMessage.has_value())
+							parameterTypeMessage = "";
+						parameterTypeMessage.value() += fmt::format(
+							"Type {} of the exponent cannot be converted to type {} of the suffix function parameter.",
+							exponent->mobileType()->toString(true),
+							suffixFunctionType->parameterTypes().at(1)->toString(true)
+						);
+					}
 				}
 			}
 			else if (suffixFunctionType->parameterTypes().size() == 1)
 			{
 				if (!literalType->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
-					parameterTypeMessage = "The type of the literal cannot be converted to the parameter of the suffix function.";
+				{
+					if(literalRationalType)
+					{
+						if (literalRationalType->mobileType())
+							parameterTypeMessage = fmt::format(
+								"The type {} of the literal cannot be converted to the type {} of the suffix function parameter.",
```

Comment:
It's not the type of the literal, it's its mobile type. I think it's confusing to mention it so I'd just omit it. The "why" should be obvious to the user by looking at the literal:

```suggestion
								"The literal is out of range of type {} accepted by the suffix function.",
```